packages/lib/src/delivery/Messages.ts

```diff
@@ -86,46 +88,39 @@ export async function incomingMessage(
     }
 }
 
-function addPostmark(
+async function addPostmark(
     { encryptedData }: EncryptionEnvelop,
     receiverEncryptionKey: string,
     deliveryServiceSigningKey: string,
-): string {
+): Promise<EncryptedPayload> {
     const postmarkWithoutSig: Omit<Postmark, 'signature'> = {
-        messageHash: ethers.utils.hashMessage(encryptedData),
+        messageHash: ethers.utils.hashMessage(stringify(encryptedData)),
         incommingTimestamp: new Date().getTime(),
     };
 
-    const signature = signPostmark(
+    const signature = await signPostmark(
         postmarkWithoutSig,
         deliveryServiceSigningKey,
     );
 
     //Encrypte the signed Postmark and return the ciphertext
-    const { ciphertext, nonce, version, ephemPublicKey } = encryptSafely({
-        publicKey: receiverEncryptionKey,
-        data: { ...postmarkWithoutSig, signature },
-        version: 'x25519-xsalsa20-poly1305',
-    });
+    const { ciphertext, nonce, ephemPublicKey } = await encryptAsymmetric(
+        receiverEncryptionKey,
+        stringify({ ...postmarkWithoutSig, signature })!,
+    );
 
-    const encryptedPostmark = stringify({
+    return {
         nonce,
-        version,
         ciphertext,
         ephemPublicKey,
-    })!;
-
-    const { hexlify, toUtf8Bytes } = ethers.utils;
-
-    return hexlify(toUtf8Bytes(encryptedPostmark));
+    };
 }
 
-const signPostmark = (p: Omit<Postmark, 'signature'>, signingKey: string) => {
+function signPostmark(
+    p: Omit<Postmark, 'signature'>,
+    signingKey: string,
+): Promise<string> {
+    log(`SING POST MARK USING ${signingKey}`);
```

Comment:
Do we want to keep this log statement? I think it's probably left from one of my PR's maybe we can remove It now.